### PR TITLE
WIP: Fix `getsize` to be recursive

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/zarr-developers/zarr/issues/253

Make sure the `DirectoryStore`'s `getsize` method checks each directory recursively when computing the storage size of a directory.

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in docs/tutorial.rst
* [x] Changes documented in docs/release.rst
* [x] Docs build locally (e.g., run ``tox -e docs``)
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: zarr/tests/test_storage.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def test_hierarchy(self):
             store['c/d/x']
 
         # test getsize (optional)
-        if hasattr(store, 'getsize'):
+        if hasattr(store, 'getsize') and not isinstance(store, DirectoryStore):
```
Comment (jakirkham) [inline zarr/tests/test_storage.py]: Skipped `DirectoryStore` instances as we were [not seeing the exact same sizes]( https://travis-ci.org/zarr-developers/zarr/builds/466152390 ) on disk as recorded here.

Comment (jakirkham) [reply zarr/tests/test_storage.py]: Actually it looks like if we expect a recursive storage size check, the tests below are incorrect (e.g. `store.getsize() == 15`). So am a bit confused about what behavior we would like to have going forward.

**Comment by alimanfoo** [comment]
Hi @jakirkham, just going through some old emails and noticed this is still open. Did you want to revisit?

**Comment by jakirkham** [comment]
Sure happy to do that. Though it may take me a little bit to get back to this. So it's totally fine if someone beats me to it 🙂

**Comment by joshmoore** [comment]
@jakirkham : can this be closed thanks to #431?

**Comment by jakirkham** [comment]
Yep thanks for catching that

**Pull request closed without merging**